corehq/apps/data_interfaces/management/commands/fix_bad_reassigned_owner_ids.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import itertools
+from six.moves import map
+
+from django.core.management import BaseCommand
+
+from casexml.apps.case.mock import CaseFactory
+from corehq.apps.domain.models import Domain
+from corehq.apps.es.users import UserES
+from corehq.apps.es.groups import GroupES, is_case_sharing
+from corehq.apps.locations.models import SQLLocation
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors, CaseAccessors
+from corehq.util.log import with_progress_bar
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'domains',
+            metavar='domain',
+            nargs='*',
+        )
+        parser.add_argument(
+            '--execute',
+            action='store_true',
+        )
+        parser.add_argument(
+            '--log_filename',
+            default='fix_bad_reassigned_owner_ids.txt',
+        )
+
+    def handle(self, domains, **options):
+        execute = options['execute']
+        log_filename = options['log_filename']
+        if not domains:
+            domains = Domain.get_all_names()
+
+        with open(log_filename, 'w') as log_file:
+            for domain in with_progress_bar(domains):
+                valid_owner_ids = get_valid_owner_ids(domain)
+                for case in get_affected_cases(domain):
+                    if case.owner_id in valid_owner_ids:
+                        continue
+
+                    matching_owner_ids = [
+                        valid_owner_id for valid_owner_id in valid_owner_ids
+                        if case.owner_id in valid_owner_id
+                    ]
+                    if len(matching_owner_ids) == 1:
+                        matched_owner_id = matching_owner_ids[0]
+                        print('%s: Assign case %s to owner %s' % (domain, case.case_id, matched_owner_id), file=log_file)
```

Comment:
Why `print` instead of `logfile.write` ?

Reply:
Because I first implemented it with ```print``` and then decided to write to a file :P